File: equipAdjust.rb
﻿#==============================================================================
# ■ スペース調整 by 莞爾の草
#==============================================================================
# 装備欄＋αのスペース数を調整します
#==============================================================================

class Window_ItemList
  #--------------------------------------------------------------------------
  # ● アイテムリストの作成
  #--------------------------------------------------------------------------
  alias __item_space_make_item_list make_item_list
  def make_item_list
    __item_space_make_item_list
    if include?(nil) && col_max > 1
      nil_size = item_max % col_max
      nil_size = col_max if nil_size.zero?
      nil_size.times { @data.push(nil) }
    end
  end
end

File: equipAdjust.js
//=============================================================================
// equipAdjust.js
//=============================================================================
/*:
 * @plugindesc adjust the space of the item list of equip screen.
 * @author Kanji the Grass
 */

/*:ja
 * @plugindesc 装備アイテムの空欄をちょうどいいように調整します。
 * @author 莞爾の草
 */

(function () {
    const _Window_ItemList_makeItemList = Window_ItemList.prototype.makeItemList;
    Window_ItemList.prototype.makeItemList = function() {
        _Window_ItemList_makeItemList.call(this);
        const col = this.maxCols();
        if (this.includes(null) && col > 1) {
            var nullSize = this.maxItems() % col;
            if (!nullSize) nullSize = col;
            for (var i = 0; i < nullSize; i++) this._data.push(null);
        }
    };
})()